src/platform/packages/shared/kbn-apm-synthtrace/src/cli/utils/start_live_data_upload.ts

```diff
@@ -24,102 +29,148 @@ export async function startLiveDataUpload({
   from: number;
   to: number;
 }) {
-  const file = runOptions.file;
+  const files = runOptions.files;
+
+  const clientsPerIndices = new Map<string, string>();
 
   const { logger, clients } = await bootstrap(runOptions);
 
-  const scenario = await getScenario({ file, logger });
-  const {
-    generate,
-    bootstrap: scenarioBootstrap,
-    teardown: scenarioTearDown,
-  } = await scenario({ ...runOptions, logger, from, to });
-
-  function startPeriodicPerfLogging() {
-    let cpuUsage = process.cpuUsage();
-
-    return setInterval(() => {
-      cpuUsage = process.cpuUsage(cpuUsage);
-      const mem = memoryUsage();
-      logger.debug(
-        `cpu time: (user: ${Math.round(cpuUsage.user / 1000)}mss, sys: ${Math.round(
-          cpuUsage.system / 1000
-        )}ms), memory: ${mb(mem.heapUsed)}/${mb(mem.heapTotal)}`
-      );
-    }, 5000);
-  }
+  Object.entries(clients).forEach(([key, client]) => {
+    if (client instanceof SynthtraceEsClient) {
+      clientsPerIndices.set(client.getAllIndices().join(','), key);
+    }
+  });
 
-  const intervalId = startPeriodicPerfLogging();
+  const scenarios = await Promise.all(
+    files.map(async (file) => {
+      const fn = await getScenario({ file, logger });
+      return fn({
+        ...runOptions,
+        logger,
+        from,
+        to,
+      });
+    })
+  );
 
-  const teardown = once(async () => {
-    if (scenarioTearDown) {
-      await scenarioTearDown(clients);
-    }
+  const stopPerformanceLogger = startPerformanceLogger({ logger });
 
-    clearInterval(intervalId);
+  const teardown = once(async () => {
+    await Promise.all(
+      scenarios.map((scenario) => {
+        if (scenario.teardown) {
+          return scenario.teardown(clients);
+        }
+      })
+    );
+
+    stopPerformanceLogger();
   });
 
   const streamManager = new StreamManager(logger, teardown);
 
-  if (scenarioBootstrap) {
-    await scenarioBootstrap(clients);
-  }
+  await Promise.all(
+    scenarios.map((scenario) => {
+      if (scenario.bootstrap) {
+        return scenario.bootstrap(clients);
+      }
+    })
+  );
 
   const bucketSizeInMs = runOptions.liveBucketSize;
-  let requestedUntil = from;
-
-  function mb(value: number): string {
-    return Math.round(value / 1024 ** 2).toString() + 'mb';
-  }
 
-  async function uploadNextBatch() {
-    const now = Date.now();
+  const workersWaitingRefresh = new Map<string, string>();
 
-    if (now > requestedUntil) {
-      const bucketCount = Math.floor((now - requestedUntil) / bucketSizeInMs);
+  function refreshIndices() {
```

Comment:
This is not the cleanest way IMO. The indices are configured per client, but we don't necessarily need to call `refresh`  on each client to refresh them. I think the es client should be independent, and the subclients should just be responsible for configuring the pipeline and targeting the indices.